core/node/state_keeper/src/keeper.rs

```diff
@@ -416,12 +418,21 @@ impl ZkSyncStateKeeper {
         )
     )]
     async fn start_next_l2_block(
-        params: L2BlockParams,
         updates_manager: &mut UpdatesManager,
         batch_executor: &mut dyn BatchExecutor<OwnedStorage>,
     ) -> anyhow::Result<()> {
-        updates_manager.push_l2_block(params);
+        updates_manager.push_l2_block();
         let block_env = updates_manager.l2_block.get_env();
+        let timestamp = updates_manager
+            .get_next_l2_block_params_or_batch_params()
+            .timestamp;
+        tracing::debug!(
+            "Initialized new L2 block #{} (L1 batch #{}) with timestamp {}",
+            updates_manager.l2_block.number + 1,
+            updates_manager.l1_batch.number,
+            display_timestamp(timestamp)
+        );
+        updates_manager.set_next_l2_block_params(None);
```

Comment:
Please use `Self:: set_l2_block_params()` here and move the log message there as discussed previously.

Reply:
I am not sure which log you are talking about but I added one in Self:: set_l2_block_params()